opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -1163,37 +1177,32 @@ namespace Opm {
-        m_messageLimits.setCommentPrintLimit(currentStep, commentPrint);
-        m_messageLimits.setCommentStopLimit(currentStep, commentStop);
-
-        int warnPrint = record.getItem("WARNING_PRINT_LIMIT").get< int >(0);
-        int warnStop = record.getItem("WARNING_STOP_LIMIT").get< int >(0);
-        m_messageLimits.setWarningPrintLimit(currentStep, warnPrint);
-        m_messageLimits.setWarningStopLimit(currentStep, warnStop);
-
-        int probPrint = record.getItem("PROBLEM_PRINT_LIMIT").get< int >(0);
-        int probStop = record.getItem("PROBLEM_STOP_LIMIT").get< int >(0);
-        m_messageLimits.setWarningPrintLimit(currentStep, probPrint);
-        m_messageLimits.setWarningStopLimit(currentStep, probStop);
-
-        int errPrint = record.getItem("ERROR_PRINT_LIMIT").get< int >(0);
-        int errStop = record.getItem("ERROR_STOP_LIMIT").get< int >(0);
-        m_messageLimits.setErrorPrintLimit(currentStep, errPrint);
-        m_messageLimits.setErrorStopLimit(currentStep, errStop);
-
-        int bugPrint = record.getItem("BUG_PRINT_LIMIT").get< int >(0);
-        int bugStop = record.getItem("BUG_STOP_LIMIT").get< int >(0);
-        m_messageLimits.setBugPrintLimit(currentStep, bugPrint);
-        m_messageLimits.setBugStopLimit(currentStep, bugStop);
+        using  set_limit_fptr = void (MessageLimits::*) (size_t , int);
+        std::vector<std::pair<std::string , set_limit_fptr>> setters = {{"MESSAGE_PRINT_LIMIT" , &MessageLimits::setMessagePrintLimit},
+                                                                        {"COMMENT_PRINT_LIMIT" , &MessageLimits::setCommentPrintLimit},
+                                                                        {"WARNING_PRINT_LIMIT" , &MessageLimits::setWarningPrintLimit},
+                                                                        {"PROBLEM_PRINT_LIMIT" , &MessageLimits::setProblemPrintLimit},
+                                                                        {"ERROR_PRINT_LIMIT"   , &MessageLimits::setErrorPrintLimit},
+                                                                        {"BUG_PRINT_LIMIT"     , &MessageLimits::setBugPrintLimit},
+                                                                        {"MESSAGE_STOP_LIMIT"  , &MessageLimits::setMessageStopLimit},
+                                                                        {"COMMENT_STOP_LIMIT"  , &MessageLimits::setCommentStopLimit},
+                                                                        {"WARNING_STOP_LIMIT"  , &MessageLimits::setWarningStopLimit},
+                                                                        {"PROBLEM_STOP_LIMIT"  , &MessageLimits::setProblemStopLimit},
+                                                                        {"ERROR_STOP_LIMIT"    , &MessageLimits::setErrorStopLimit},
+                                                                        {"BUG_STOP_LIMIT"      , &MessageLimits::setBugStopLimit}};
+
```

Comment:
Nifty solution! A couple of points:

Just for fun, there's not *that* much of a vector (implies dynamic allocation) for this; by using auto (initializer_list) or just a raw or std::array we'd save an allocation. Finally, there's a cool thing in functional called mem_fn. Consider:

```
    void Schedule::handleMESSAGES( const DeckKeyword& keyword, size_t currentStep) {
        const auto& record = keyword.getRecord(0);
        using set_limit = decltype( std::mem_fn( &MessageLimits::setMessagePrintLimit ) );
        std::pair< std::string, set_limit > setters[] = {
            {"MESSAGE_PRINT_LIMIT", std::mem_fn( &MessageLimits::setMessagePrintLimit ) },                                                                                                                                                                                        
            {"COMMENT_PRINT_LIMIT", std::mem_fn( &MessageLimits::setCommentPrintLimit ) },
            {"WARNING_PRINT_LIMIT", std::mem_fn( &MessageLimits::setWarningPrintLimit ) },
            {"PROBLEM_PRINT_LIMIT", std::mem_fn( &MessageLimits::setProblemPrintLimit ) },
            {"ERROR_PRINT_LIMIT"  , std::mem_fn( &MessageLimits::setErrorPrintLimit ) },
            {"BUG_PRINT_LIMIT"    , std::mem_fn( &MessageLimits::setBugPrintLimit ) },
            {"MESSAGE_STOP_LIMIT" , std::mem_fn( &MessageLimits::setMessageStopLimit ) }, 
            {"COMMENT_STOP_LIMIT" , std::mem_fn( &MessageLimits::setCommentStopLimit ) }, 
            {"WARNING_STOP_LIMIT" , std::mem_fn( &MessageLimits::setWarningStopLimit ) },
            {"PROBLEM_STOP_LIMIT" , std::mem_fn( &MessageLimits::setProblemStopLimit ) },
            {"ERROR_STOP_LIMIT"   , std::mem_fn( &MessageLimits::setErrorStopLimit ) },
            {"BUG_STOP_LIMIT"     , std::mem_fn( &MessageLimits::setBugStopLimit ) }
        };    

        for (const auto& pair : setters) {
            const auto& item = record.getItem( pair.first );
            if (item.defaultApplied(0)) continue;

            int value = item.get<int>(0);
            pair.second( this->m_messageLimits, currentStep, item.get< int >( 0 ) );
        }     
    }
```

Pretty cool!

This is, however, ready whenever.